core/creator/src/main/java/io/quarkus/creator/phase/nativeimage/NativeImagePhase.java

```diff
@@ -313,6 +313,46 @@ public void provideOutcome(AppCreator ctx) throws AppCreatorException {
             // E.g. "/usr/bin/docker run -v {{PROJECT_DIR}}:/project --rm quarkus/graalvm-native-image"
             nativeImage = new ArrayList<>();
             Collections.addAll(nativeImage, containerRuntime, "run", "-v", outputDir.toAbsolutePath() + ":/project:z", "--rm");
+            if (IS_LINUX & "docker".equals(containerRuntime)) {
+                try {
+                    BufferedReader reader;
+                    StringBuilder builder;
+                    String uid = null;
+                    String gid = null;
+
+                    ProcessBuilder idPB = new ProcessBuilder().command("id", "-ur");
+                    Process process = idPB.start();
+
+                    if (process.waitFor() == 0) {
+                        reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+                        builder = new StringBuilder();
+                        String line;
+                        while ((line = reader.readLine()) != null) {
+                            builder.append(line);
+                        }
+                        uid = builder.toString();
+                    }
+
+                    idPB = new ProcessBuilder().command("id", "-gr");
+                    process = idPB.start();
+
+                    if (process.waitFor() == 0) {
+                        reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
```

Comment:
I see this is using a `java.lang.UNIXProcess.ProcessPipeInputStream`, we should probably close those.

(same above .. maybe have a nice helper method rather than duplicating the code?)

Reply:
I am closing all the streams now, and have extracted the duplicated code to a static helper method